### PR TITLE
Patch libsolv to not throw on subdir mismatch

### PR DESCRIPTION
**Pull request by johnhany97**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Context: This PR is a step towards a proposed solution to fix https://github.com/mamba-org/mamba/issues/2363. The issue is manifesting as a global outage for any workflow that's using mamba/micromamba due to certain channels including packages that declare a subdir that's different from the repodata.json subdir it's found in.

### CODE REVIEW

**Comment by johnhany97** [comment]
@conda-forge-admin, please rerender

File: .scripts/build_steps.sh [hunk 1]
```diff
@@ -32,9 +32,9 @@ pkgs_dirs:
 CONDARC
 
 
-mamba install --update-specs --yes --quiet --channel conda-forge \
+conda install --update-specs --yes --quiet --channel conda-forge \
     conda-build pip boa conda-forge-ci-setup=3
-mamba update --update-specs --yes --quiet --channel conda-forge \
+conda install --update-specs --yes --quiet --channel conda-forge \
```
Comment (johnhany97) [inline .scripts/build_steps.sh:37]: This is temporary and should be reverted once https://github.com/conda-forge/docker-images/pull/236 is released.

Comment (Zaubentrucker) [reply .scripts/build_steps.sh:37]: @johnhany97 
Just to be sure, was it on purpose to change this to `conda install` instead of `conda update` ?

Comment (johnhany97) [reply .scripts/build_steps.sh:37]: I changed to `conda install` because `conda update` failed and asked me to run `install` instead.

**Review by johnhany97 — COMMENTED** [review]
Thank you @wolfv for carrying this through!

**Pull request merged**